ARCs/arc-0015.md

```diff
@@ -0,0 +1,241 @@
+---
+arc: 15
+title: Encrypted Short Messages
+description: Scheme for encryption/decryption that allows for private messages.
+author: Stéphane Barroso (@sudoweezy), Paweł Pierścionek (@urtho)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/*
+status: Draft
+type: Standards Track
+category: ARC
+created: 2022-11-21
+requires: 4
+---
+
+# Encrypted Short Messages 
+
+## Abstract
+
+The goal of this convention is to have a standard way for block explorers, wallets, exchanges, marketplaces, and more generally, client software to send, read & delete short encrypted messages.
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+> Comments like this are non-normative.
+
+### Account's message Application
+
+To receive a message, an Account **MUST** create an application that follows this convention:
+
+- A Local State named `public_key` **MUST** contain an `NACL Public Key (Curve 25519)`
+
+- A Box `inbox` where:
+  - Keys is an ABI encoded of the tuple `(address,uint64)` containing the address of the sender and the round when the message is sent
+  - Values are encoded **text**
+
+> With this design, for each round, the sender can only write one message per round.
+> For the same round, an account can receive multiple messages if distinct sender sends them
+
+### ABI Interface
+
+The associated smart contract **MUST** implement the following ABI interface:
+```json
+{
+  "name": "ARC_0015",
+  "desc": "Interface for an encrypted messages application",
+  "methods": [
+    {
+      "name": "write",
+      "desc": "Write encrypted text to the box inbox",
+      "args": [
+        { "type": "byte[]", "name": "text", "desc": "Encryted text provided by the sender." }
+      ],
+      "returns": { "type": "void" }
+    },
+    {
+      "name": "remove",
```

Comment:
I'm not sure about this function:
nothing can be removed from the blockchain.

If we think it's important, I would recommend adding a warning that people can still retrieve "Removed" messages. There is no way to prevent that.

Reply:
I think it is still important to have a way for "unlocking" Algo used for the Box.
Nevertheless, I will add the Warning!